core/spec/models/spree/order/checkout_spec.rb

```diff
@@ -408,14 +408,32 @@ def assert_state_changed(order, from, to)
         context 'when there is at least one valid payment' do
           let(:payment_state) { 'checkout' }
 
-          before do
-            expect(order).to receive(:process_payments!).once { true }
+          context 'line_items are in stock' do
+            before do
+              expect(order).to receive(:process_payments!).once { true }
+            end
+
+            it "transitions to complete" do
+              order.next!
+              assert_state_changed(order, 'payment', 'complete')
+              expect(order.state).to eq('complete')
+            end
           end
 
-          it "transitions to complete" do
-            order.next!
-            assert_state_changed(order, 'payment', 'complete')
-            expect(order.state).to eq('complete')
+          context 'line_items are not in stock' do
+            before do
+              expect(order).to receive(:ensure_line_items_are_in_stock).once { false }
+            end
+
+            it 'should not receive process_payments!' do
+              expect(order).not_to receive(:process_payments!)
+              order.next
```

Comment:
@Mafi88 `order.next` is needed here.
when we will call next then only it will be calling the `before_transition to: :complete` callbacks.